File: java/sourceafis/src/sourceafis/templates/TemplateBuilder.java
package sourceafis.templates;

import java.util.ArrayList;
import java.util.List;

import sourceafis.general.Calc;
 
public class TemplateBuilder {
	 
	public List<Minutia> minutiae = new ArrayList<Minutia>();

	/*
	public static class Minutia {
		public Point Position=new Point(0,0);
		public byte Direction;
		public MinutiaType Type;
	}
	*/
	public int originalDpi;
	public int originalWidth;
	public int originalHeight;

	public int getStandardDpiWidth() {
		return Calc.DivRoundUp(originalWidth * 500, originalDpi);
	}

	public void setStandardDpiWidth(int value) {
		originalWidth = value * originalDpi / 500;
	}

	public int getStandardDpiHeight() {
		return Calc.DivRoundUp(originalHeight * 500, originalDpi);
	}

	public void setStandardDpiHeight(int value) {
		originalHeight = value * originalDpi / 500;
	}
 

}

File: SourceAFIS/Templates/TemplateBuilder.cs
using System;
using System.Collections.Generic;
using System.Text;
using SourceAFIS.General;
using SourceAFIS.Dummy;

namespace SourceAFIS.Templates
{
    public sealed class TemplateBuilder : ICloneable
    {
        public enum MinutiaType
        {
            Ending = 0,
            Bifurcation = 1,
            Other = 2
        }

        public class Minutia
        {
            public Point Position;
            public byte Direction;
            public MinutiaType Type;
        }

        public int OriginalDpi;
        public int OriginalWidth;
        public int OriginalHeight;

        public int StandardDpiWidth
        {
            get { return Calc.DivRoundUp(OriginalWidth * 500, OriginalDpi); }
            set { OriginalWidth = value * OriginalDpi / 500; }
        }
        public int StandardDpiHeight
        {
            get { return Calc.DivRoundUp(OriginalHeight * 500, OriginalDpi); }
            set { OriginalHeight = value * OriginalDpi / 500; }
        }

        public List<Minutia> Minutiae = new List<Minutia>();

        public TemplateBuilder Clone()
        {
            return new Template(this).ToTemplateBuilder();
        }

        object ICloneable.Clone() { return Clone(); }
    }
}
